Sources/Bookmark/NYPLBookmarkSpec.swift

```diff
@@ -174,24 +197,24 @@ struct NYPLBookmarkSpec {
   let motivation: Motivation
   let target: Target
 
-  init(id: String? = nil,
-       time: Date,
-       device: String,
-       bodyOthers: [String: String]? = nil,
-       motivation: Motivation,
-       bookID: String,
-       selectorValue: String) {
+  public init(id: String? = nil,
+              time: String,
```

Comment:
wondering if we could keep the stronger `Date` type as input. This allowed us to format the date in the RFC 3339 format which as far as I remember was a strict requirement. If we loosen the data type here, we'll need to make this initializer a `init?` in case the string is not in the RFC 3339 format.  In any case, I'm probably missing something.

Reply:
I did not know the RFC 3339 format is a strict requirement. In such case, I will rewrite the RFC 3339 related function in swift and migrate it to this repo. Will keep you posted.